pyiron_base/project/generic.py

```diff
@@ -512,34 +512,53 @@ def inspect(self, job_specifier):
         """
         return self.load(job_specifier=job_specifier, convert_to_object=False)
 
-    def iter_jobs(self, path=None, recursive=True, convert_to_object=True, status=None, job_type=None, progress=True):
+    def get_filtered_job_ids(self, recursive: bool = True, **kwargs: dict) -> list:
+        """
+        Get a list of job ids in a project based on matching values from any column in the project database
+
+        Args:
+            recursive (bool): True if entries from subprojects are considered
+            **kwargs (dict): Optional arguments for filtering with keys matching the project database column name
+                            (eg. status="finished")
+
+        Returns:
+            list: List of job IDs
+        """
+        if len(kwargs) == 0:
+            return self.get_job_ids(recursive=recursive)
+        df = self.job_table(recursive=recursive)
+        if df.empty:
+            return []
+        mask = np.ones_like(df.index, dtype=bool)
+        db_columns = self.get_db_columns()
+        for key in kwargs.keys():
+            if key not in db_columns:
+                raise ValueError("Column name {} does not exist in the project database!")
+        for key, val in kwargs.items():
+            if val is None:
+                mask &= df[key].isnull()
+            else:
+                mask &= df[key] == val
+        if not mask.any():
+            return []
+        return df[mask]["id"].to_list()
+
+    def iter_jobs(self, path=None, recursive=True, convert_to_object=True, progress=True, **kwargs):
```

Comment:
```suggestion
    def iter_jobs(self, path=None, recursive: bool=True, convert_to_object: bool=True, progress: bool=True, **kwargs) -> Iterator[GenericJob]:
```
~Did we loose the `path` parameter somewhere?~ Nevermind, github hid the rest of the method and I was not smart enough to click on expand.